cate/core/opimpl.py

```diff
@@ -71,6 +72,57 @@ def normalize_impl(ds: xr.Dataset) -> xr.Dataset:
     return ds
 
 
+def _normalize_zonal_lat_lon(ds: xr.Dataset) -> xr.Dataset:
+    """
+    In case that the dataset only contains lat_centers and is a zonal mean dataset,
+    the longitude dimension created and filled with the variable value of certain latitude.
+    :param ds: some xarray dataset
+    :return: a normalized xarray dataset
+    """
+
+    if 'latitude_centers' not in ds.coords:
+        return ds
+    if 'lon' in ds.coords:
+        return ds
+    ds_zonal = ds.copy()
+    resolution = (ds.latitude_centers.values[1] - ds.latitude_centers.values[0])
+    ds_zonal = ds_zonal.assign_coords(lon=[i + (resolution / 2) for i in np.arange(-180.0, 180.0, resolution)])
+
+    for var in ds_zonal.variables:
+        if var not in ds_zonal.coords:
+            if sorted([dim for dim in ds_zonal[var].dims]) == sorted([coord for coord in ds.coords]):
+                ds_zonal[var] = xr.concat([ds_zonal[var] for _ in ds_zonal.lon], 'lon')
+                ds_zonal[var]['lon'] = ds_zonal.lon
+    ds_zonal = ds_zonal.assign_coords(lat=ds.latitude_centers.values)
+    ds_zonal = ds_zonal.rename_dims({'latitude_centers': 'lat'})
+    ds_zonal = ds_zonal.drop('latitude_centers')
+    ds_zonal = ds_zonal.transpose(..., 'lat', 'lon')
+
+    has_lon_bnds = 'lon_bnds' in ds_zonal.coords or 'lon_bnds' in ds_zonal
+    if not has_lon_bnds:
+        ds_zonal = ds_zonal.assign_coords(
+            lon_bnds=xr.DataArray([[i - (resolution / 2), i + (resolution / 2)] for i in ds_zonal.lon.values],
+                                  dims=['lon', 'bnds']))
+    has_lat_bnds = 'lat_bnds' in ds_zonal.coords or 'lat_bnds' in ds_zonal
+    if not has_lat_bnds:
+        ds_zonal = ds_zonal.assign_coords(
+            lat_bnds=xr.DataArray([[i - (resolution / 2), i + (resolution / 2)] for i in ds_zonal.lat.values],
+                                  dims=['lat', 'bnds']))
+
+    ds_zonal.lon.attrs['bounds'] = 'lon_bnds'
+    ds_zonal.lon.attrs['long_name'] = 'longitude'
+    ds_zonal.lon.attrs['standard_name'] = 'longitude'
+    ds_zonal.lon.attrs['units'] = 'degrees_east'
+
+    ds_zonal.lat.attrs['bounds'] = 'lat_bnds'
+    ds_zonal.lat.attrs['long_name'] = 'latitude'
+    ds_zonal.lat.attrs['standard_name'] = 'latitude'
+    ds_zonal.lat.attrs['units'] = 'degrees_north'
+
+    ds = ds_zonal.copy()
```

Comment:
Why copy again? You already did it once above, and even then I didn't think it was necessary.